cmd/api/main.go

```diff
@@ -381,19 +388,45 @@ func HandleRebuildPackage(rw http.ResponseWriter, req *http.Request) {
 		}
 	}
 	var manualStrategy, strategy rebuild.Strategy
+	var buildDefLoc rebuild.Location
 	ireq := schema.InferenceRequest{
 		Ecosystem: rbreq.Ecosystem,
 		Package:   rbreq.Package,
 		Version:   rbreq.Version,
 	}
-	if rbreq.StrategyOneof != nil {
-		manualStrategy, err = rbreq.StrategyOneof.Strategy()
+	if rbreq.StrategyFromRepo {
+		repo, err := uri.CanonicalizeRepoURI(*buildDefRepo)
 		if err != nil {
-			err := errors.Wrap(err, "parsing provided strategy")
+			err := errors.Wrap(err, "canonicalizing build def repo")
 			log.Println(err)
 			http.Error(rw, err.Error(), 400)
 			return
 		}
+		dir := path.Clean(*buildDefRepoDir)
+		defs, err := builddef.NewBuildDefinitionSetFromGit(&builddef.GitBuildDefinitionSetOptions{
+			CloneOptions: git.CloneOptions{
+				URL:           repo,
+				ReferenceName: plumbing.Main,
+				Depth:         1,
+				NoCheckout:    true,
+			},
+			RelativePath: dir,
+			// TODO: Limit this further to only the target's path we want.
+			SparseCheckoutDirs: []string{dir},
+		})
+		if err != nil {
+			err = errors.Wrap(err, "parsing provided strategy")
```

Comment:
Is this error message correct? IIUC err != nil means the git build def set failed to be constructed, but no strategies have been parsed yet?